src/components/ha-selector/ha-selector-text.ts

```diff
@@ -75,7 +75,10 @@ export class HaTextSelector extends LitElement {
     }
     if (value === "" && !this.required) {
       value = undefined;
+    } else if (this.selector.text.type === "number") {
```

Comment:
It is weird btw to use the text selector with type number, use the number selector